test/cppsuite/src/main/test.cpp

```diff
@@ -36,37 +36,53 @@ namespace test_harness {
 test::test(const test_args &args) : _args(args)
 {
     _config = new configuration(args.test_name, args.test_config);
-    _metrics_monitor =
-      new metrics_monitor(args.test_name, _config->get_subconfig(METRICS_MONITOR), _database);
     _timestamp_manager = new timestamp_manager(_config->get_subconfig(TIMESTAMP_MANAGER));
     _workload_manager = new workload_manager(
       _config->get_subconfig(WORKLOAD_MANAGER), this, _timestamp_manager, _database);
     _thread_manager = new thread_manager();
 
+    /* Only create the metrics monitor if enabled. */
+    auto metrics_monitor_cfg = _config->get_subconfig(METRICS_MONITOR);
+    if (metrics_monitor_cfg->get_bool(ENABLED))
+        _metrics_monitor = new metrics_monitor(args.test_name, metrics_monitor_cfg, _database);
+    else
+        delete metrics_monitor_cfg;
+
     _database.set_timestamp_manager(_timestamp_manager);
     _database.set_create_config(
       _config->get_bool(COMPRESSION_ENABLED), _config->get_bool(REVERSE_COLLATOR));
 
-    /*
-     * Ordering is not important here, any dependencies between components should be resolved
-     * internally by the components.
-     */
-    _components = {_workload_manager, _timestamp_manager, _metrics_monitor};
+    /* Update the component list with the enabled ones. */
+    _components.push_back(_workload_manager);
+    if (_timestamp_manager->enabled())
+        _components.push_back(_timestamp_manager);
+    if (_metrics_monitor != nullptr && _metrics_monitor->enabled())
+        _components.push_back(_metrics_monitor);
 }
 
 void
 test::init_operation_tracker(operation_tracker *op_tracker)
 {
     delete _operation_tracker;
+    std::unique_ptr<configuration> operation_tracker_cfg(_config->get_subconfig(OPERATION_TRACKER));
+    bool tracking_enabled = operation_tracker_cfg->get_bool(ENABLED);
     if (op_tracker == nullptr) {
         /* Fallback to default behavior. */
         op_tracker = new operation_tracker(_config->get_subconfig(OPERATION_TRACKER),
           _config->get_bool(COMPRESSION_ENABLED), *_timestamp_manager);
+    } else {
+        /*
+         * If a custom tracker has been given, make sure it has been enabled in the test
+         * configuration.
+         */
+        testutil_assert(tracking_enabled);
     }
     _operation_tracker = op_tracker;
```

Comment:
I suggest moving the `delete` on line 66 to just before this assignment. While I think ultimately this will be simplified by the use of smart pointers, for the time being keeping these ops close together makes it easier to do the accounting, and `_operation_tracker` is not used until this point.